bundles/org.openhab.binding.upnpcontrol/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,172 @@
+	<thing-type id="upnprenderer">
+		<label>UPnPRenderer</label>
+		<description>UPnP AV Renderer</description>
+		<channels>
+			<channel id="volume" typeId="volume"/>
+			<channel id="mute" typeId="mute"/>
+			<channel id="control" typeId="control"/>
+			<channel id="stop" typeId="stop"/>
+			<channel id="title" typeId="title"/>
+			<channel id="album" typeId="album"/>
+			<channel id="albumart" typeId="albumart"/>
+			<channel id="creator" typeId="creator"/>
+			<channel id="artist" typeId="artist"/>
+			<channel id="publisher" typeId="publisher"/>
+			<channel id="genre" typeId="genre"/>
+			<channel id="tracknumber" typeId="tracknumber"/>
+			<channel id="trackduration" typeId="trackduration"/>
+			<channel id="trackposition" typeId="trackposition"/>
+		</channels>
+		<config-description>
+			<parameter name="udn" type="text" required="true">
+				<label>Unique Device Name</label>
+				<description>The UDN identifies the UPnP Renderer</description>
+			</parameter>
+		</config-description>
+
+	</thing-type>
+	<thing-type id="upnpserver">
+		<label>UPnPServer</label>
+		<description>UPnP AV Server</description>
+		<channels>
+			<channel id="upnprenderer" typeId="upnprenderer"/>
+			<channel id="currentid" typeId="currentid"/>
+			<channel id="browse" typeId="browse"/>
+			<channel id="search" typeId="search"/>
+		</channels>
+		<config-description>
+			<parameter name="udn" type="text" required="true">
+				<label>Unique Device Name</label>
+				<description>The UDN identifies the UPnP Media Server</description>
+			</parameter>
+			<parameter name="filter" type="boolean" required="false">
+				<label>Filter Content</label>
+				<description>Only list content which is playable on the selected renderer</description>
+				<default>false</default>
+				<advanced>false</advanced>
+			</parameter>
+			<parameter name="sortcriteria" type="text" required="false">
+				<label>Sort Criteria</label>
+				<description>Sort criteria for the titles in the selection list and when sending for playing to a renderer. The
+					criteria are defined in UPnP sort criteria format. Examples: +dc:title, -dc:creator, +upnp:album. Supported sort
+					criteria will depend on the media server</description>
+				<default>+dc:title</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Channel Types -->
+	<channel-type id="volume">
+		<item-type>Dimmer</item-type>
+		<label>Volume</label>
+		<description>Set or get the master volume</description>
+		<category>SoundVolume</category>
+	</channel-type>
+	<channel-type id="mute">
+		<item-type>Switch</item-type>
+		<label>Mute</label>
+		<description>Set or get the mute state</description>
+	</channel-type>
+	<channel-type id="control">
+		<item-type>Player</item-type>
+		<label>Control</label>
+		<description>Control the player, e.g. start/stop/next/previous/ffward/rewind</description>
+		<category>Player</category>
+	</channel-type>
+	<channel-type id="stop">
+		<item-type>Switch</item-type>
```

Comment:
Shouldn't this be a trigger channel? It's not for keeping state, just to initiate a stop?

Reply:
I am not sure about this. I did the same as in the Sonos binding. I also saw this in the Squeezebox binding, and I believe most bindings modelling a player type of device do the same. You still would want to attach an item to it to model a button in a UI. So far, I have only seen examples of trigger channels directly used from rules, without items attached to it.

Reply:
i think you are right. It's mostly used for triggers from devices to pass to a binding.